### PR TITLE
Add API functions to initialize Haveno account

### PR DESCRIPTION
**Pull request by muscleman**
commit 54b74fcda76c30d1947e5e0cc2f312ef0dca145f
Author: muscleman <musclesonvacation@hotmail.com>
Date:   Sun Nov 21 15:09:24 2021 -0600

    WIP

commit f6079499731fb6b7e83e8d49893f3e5d6e9f2452
Author: muscleman <musclesonvacation@hotmail.com>
Date:   Sat Nov 20 20:01:23 2021 -0600

    stuck at restoreAccount

commit 6546534ceec333eef2d6a7bbe23f2d17a9e7a62b
Author: muscleman <musclesonvacation@hotmail.com>
Date:   Thu Nov 18 10:40:25 2021 -0600

    Added AccountClient

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,21 +1,29 @@
 import React from 'react';
 import logo from './logo.png';
+
+import { saveAs } from 'file-saver'
```
Comment (woodser) [inline src/App.tsx:4]: Let's not change App.tsx. Everything should be tested in HavenoDaemon.test.ts.

File: src/HavenoDaemon.ts [hunk 2]
```diff
@@ -1,6 +1,23 @@
...
+import {GetVersionClient, PriceClient, WalletsClient, OffersClient, PaymentAccountsClient, TradesClient, AccountClient} from './protobuf/GrpcServiceClientPb';
+import {GetVersionRequest, GetVersionReply, MarketPriceRequest, MarketPriceReply, MarketPricesRequest, MarketPricesReply, MarketPriceInfo, GetBalancesRequest, GetBalancesReply, XmrBalanceInfo, GetOffersRequest, GetOffersReply, OfferInfo, GetPaymentAccountsRequest, GetPaymentAccountsReply, CreateCryptoCurrencyPaymentAccountRequest, CreateCryptoCurrencyPaymentAccountReply, CreateOfferRequest, CreateOfferReply, CancelOfferRequest, TakeOfferRequest, TakeOfferReply, TradeInfo, GetTradeRequest, GetTradeReply, GetNewDepositSubaddressRequest, GetNewDepositSubaddressReply, ConfirmPaymentStartedRequest, ConfirmPaymentReceivedRequest,
```
Comment (woodser) [inline src/HavenoDaemon.ts:3]: One line.

File: src/HavenoDaemon.ts [hunk 3]
```diff
@@ -33,6 +51,7 @@ class HavenoDaemon {
     this._paymentAccountsClient = new PaymentAccountsClient(this._url);
     this._offersClient = new OffersClient(this._url);
     this._tradesClient = new TradesClient(this._url);
+	this._accountClient = new AccountClient(this._url);
```
Comment (woodser) [inline src/HavenoDaemon.ts:54]: Inconsistent whitespace.

File: src/HavenoDaemon.ts [hunk 4]
```diff
@@ -311,6 +330,148 @@ class HavenoDaemon {
       });
     });
   }
+
+  ///////////////////////////////////////////////
+  // Account
+  ///////////////////////////////////////////////
+
+  /**
+   * Determine if an Account Exists.
```
Comment (woodser) [inline src/HavenoDaemon.ts:339]: Account Exists should be lowercase.

File: src/HavenoDaemon.ts [hunk 5]
```diff
@@ -311,6 +330,148 @@ class HavenoDaemon {
...
+    return new Promise(function(resolve, reject) {
+      that._accountClient.accountExists(new AccountExistsRequest(), {password: that._password}, function(err: grpcWeb.RpcError, response: AccountExistsReply) {
+        if (err) reject(err);
+        else resolve(response.getAccountExists());
+      });
+    });
+  }
+
+  /**
+   * Determine if an Account is open.
```
Comment (woodser) [inline src/HavenoDaemon.ts:354]: lowercase account

File: src/HavenoDaemon.ts [hunk 6]
```diff
@@ -311,6 +330,148 @@ class HavenoDaemon {
...
+    let that = this;
+    return new Promise(function(resolve, reject) {
+      that._accountClient.closeAccount(new CloseAccountRequest(), {password: that._password}, function(err: grpcWeb.RpcError, response: CloseAccountReply) {
+        if (err) reject(err);
+        else resolve();
+      });
+    });
+  }
+
+  async backupAccount(): Promise<Buffer> {
```
Comment (woodser) [inline src/HavenoDaemon.ts:412]: Should return `Attachment` with file name and bytes.

**Review by woodser — CHANGES_REQUESTED** [review]

File: src/HavenoDaemon.ts (hunk 2 above)
Comment (muscleman) [reply src/HavenoDaemon.ts:3]: sure thing, not my style but hey.

File: src/HavenoDaemon.ts (hunk 4 above)
Comment (muscleman) [reply src/HavenoDaemon.ts:339]: done

File: src/HavenoDaemon.ts (hunk 6 above)
Comment (muscleman) [reply src/HavenoDaemon.ts:412]: changed

File: src/HavenoDaemon.ts (hunk 3 above)
Comment (muscleman) [reply src/HavenoDaemon.ts:54]: fixed

File: src/App.tsx (hunk 1 above)
Comment (muscleman) [reply src/App.tsx:4]: i'll rip it out after i'm done with the backend. i like to click and step through the code

**Comment by erciccione** [comment]
This seems to be a duplicate of #176

**Comment by woodser** [comment]
This is the frontend of account management, while [#176](https://github.com/haveno-dex/haveno/pull/176) is the backend.

**Comment by muscleman** [comment]
yeah, i wondered why you deleted it

**Comment by erciccione** [comment]
Sorry, my bad. Haven't noticed i was on the other repo :)

**Pull request closed without merging**